indico/modules/search/controllers.py

```diff
@@ -40,17 +40,23 @@ def _process(self):
 
 
 class RHAPISearch(RH):
+    """API for searching across all records with the current search provider
+
+    Besides pagination, filters or placeholders may be passed as query parameters.
+    Since `type` may be a list, the results from the search provider are not mixed with
+    the InternalSearch.
+    """
     @use_kwargs({
         'page': fields.Int(missing=1),
         'q': fields.String(required=True),
+        'type': fields.List(EnumField(SearchTarget), missing=None)
     }, location='query', unknown=INCLUDE)
-    @use_kwargs({'type': EnumField(SearchTarget)}, location='view_args')
-    def _process(self, type, page, q, **params):
+    def _process(self, page, q, type, internal, **params):
         search_provider = get_search_provider()
-        if not search_provider or type == SearchTarget.category:
+        if not search_provider or SearchTarget.category in type:
```

Comment:
Maybe this is clearer since you won't get useful results when you pass multiple types including `category`?

```suggestion
        if not search_provider or type == [SearchTarget.category]:
```

Reply:
In this case, it may pass them through the actual provider which should have no entries either, but who knows.
Anyway, I find it equally reasonable.